### PR TITLE
Show perf comparsion of CSV loaders across DBs

### PR DESCRIPTION
**Pull request by cpcloud**
- [x] pandas `read_csv` + `to_sql`
- [x] Database to CSV perf
- [x] ~~`Hive` + the Parquet format~~, will do this later
- [x] Reorganize the narrative
- [x] ~~redshift gzip and raw text~~, put as a bunch of `TODO`s in the doc

### CODE REVIEW

**Comment by mrocklin** [comment]
This is awesome.  You should blog this (or at least tweet out the doc).  Things I'd enjoy seeing

1.  SQL->CSV calls alongside current CSV->SQL %time calls
2.  Performance using pandas or straight python (although I note the TODO) just to give context.  Also, maybe reverse the order so that pandas is first?  That sort of frames the problem as "Look, you would do it this way and it would work, but it would take ten hours."  
3.  There is a nice narrative with pandas -> pg_bulkload -> odo, especially if you show the command to use pg_bulkload.  That way we get the narrative "Pandas works, but use native loaders; they're fast." followed by "Odo leverages native solutions from python."

**Comment by cpcloud** [comment]
Going to rip off the bug fixes here into a separate PR

**Comment by cpcloud** [comment]
sqlite3 timing on the way, merging in the meantime

File: docs/source/perf.rst [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+bringing the total loading time to::
+
+  .. code-block:: python
+
+     >>> 175 * 4 / 60.0  # doctest: +ELLIPSIS
+     11.66...
+
+11.66 **hours**!
+
+Nearly *12* hours to insert 175 million rows into a postgresql database. The
```
Comment (mrocklin) [inline docs/source/perf.rst]: should this be sqlite?

Comment (cpcloud) [inline docs/source/perf.rst]: nope i used postgres for this

File: docs/source/perf.rst [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+`````````````````````
+
+.. code-block:: python
+
+   In [1]: %time t = odo('all.csv', 'mysql+pymysql://localhost/test::nyc')
+   CPU times: user 1.32 s, sys: 304 ms, total: 1.63 s
+   Wall time: 20min 49s
+
+* READS: ~30 MB/s
+* WRITES: ~150 MB/s
```
Comment (mrocklin) [inline docs/source/perf.rst]: this number seems off

Comment (cpcloud) [inline docs/source/perf.rst]: it could be a typo and i meant 50, i'll rerun this one

**Pull request merged**